From f09dd4db511247a286017d3c5e715f00cc63cfd2 Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Fri, 26 Jun 2015 19:51:20 -0500
Subject: [PATCH 38/86] arch/arm64: Temporary Work-Around for XLF Simulation

Bit 17 of the GICD TYPER register is set in simulation;
it shouldn't be. Ignore it for now.

Signed-off-by: John Jacques <john.jacques@intel.com>
[Original patch is from linux-yocto-4.1 axxia/base branch.]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/irqchip/irq-gic-v3-its.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/irqchip/irq-gic-v3-its.c b/drivers/irqchip/irq-gic-v3-its.c
index 9a791dd..5731ae3 100644
--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -1598,14 +1598,14 @@ int its_init(struct device_node *node, struct rdists *rdists,
 		its_probe(np, parent_domain);
 	}
 
+	gic_rdists = rdists;
+	gic_root_node = node;
+
 	if (list_empty(&its_nodes)) {
 		pr_warn("ITS: No ITS available, not enabling LPIs\n");
 		return -ENXIO;
 	}
 
-	gic_rdists = rdists;
-	gic_root_node = node;
-
 	its_alloc_lpi_tables();
 	its_lpi_init(rdists->id_bits);
 
-- 
1.7.5.4

